File: backend/internal/service/admin_permissions.go
package service

type AdminPermission string

const (
	AdminPermissionDashboardRead AdminPermission = "admin.dashboard.read"
	AdminPermissionOpsRead       AdminPermission = "admin.ops.read"
	AdminPermissionUsageRead     AdminPermission = "admin.usage.read"
)

func CanAccessAdminPanel(role string) bool {
	return role == RoleAdmin || role == RoleOperator
}

func HasAdminPermission(role string, permission AdminPermission) bool {
	if role == RoleAdmin {
		return true
	}
	if role != RoleOperator {
		return false
	}
	switch permission {
	case AdminPermissionDashboardRead, AdminPermissionOpsRead, AdminPermissionUsageRead:
		return true
	default:
		return false
	}
}

File: frontend/src/utils/adminPermissions.ts
import type { AdminPermission, UserRole } from '@/types'

const operatorPermissions = new Set<AdminPermission>([
  'admin.dashboard.read',
  'admin.ops.read',
  'admin.usage.read'
])

export function canAccessAdminPanel(role?: UserRole | null): boolean {
  return role === 'admin' || role === 'operator'
}

export function hasAdminPermission(role: UserRole | null | undefined, permission: AdminPermission): boolean {
  if (role === 'admin') return true
  return role === 'operator' && operatorPermissions.has(permission)
}

export function defaultHomePath(role?: UserRole | null): string {
  return canAccessAdminPanel(role) ? '/admin/dashboard' : '/dashboard'
}
